index.js

```diff
@@ -5,15 +5,35 @@ const through = require('through2');
 const debug = require('debug')('tcp-proxy');
 const EventEmitter = require('events').EventEmitter;
 
-function genThrough(interceptor) {
+function genThrough(interceptor, connection) {
+
   return through.obj(function(chunk, enc, done) {
-    const result = interceptor(chunk, enc);
+
+    const context = {
+      client: {
+        ip: connection.client.address().address,
+        port: connection.client.address().port,
+      },
+      server: {
+        ip: connection.server.address().address,
+        port: connection.server.address().port,
+      },
+      self: {
+        ip: connection.forwardHost,
+        port: connection.forwardPort,
```

Comment:
Why don't just pass through `connection.client` and `connection.server` ?

Reply:
This is because there is more to the equation than just the client and server, and that's the proxy information itself, and this can't be forwarded with just the client and server

Reply:
I don't think so, we can get any proxy information we want from `client` and `server`, why we just forwards the `ip` and `port` ?